packages/babylon/src/tokenizer/index.js

```diff
@@ -437,6 +437,32 @@ export default class Tokenizer extends LocationParser {
     }
   }
 
+  readToken_interpreter(): boolean {
+    if (this.state.pos !== 0 || this.state.input.length < 2) return false;
+
+    const start = this.state.pos;
+    this.state.pos += 1;
+
+    let ch = this.input.charCodeAt(this.state.pos);
+    if (ch !== charCodes.exclamationMark) return false;
+
+    while (
+      ch !== charCodes.lineFeed &&
+      ch !== charCodes.carriageReturn &&
+      ch !== charCodes.lineSeparator &&
+      ch !== charCodes.paragraphSeparator &&
+      ++this.state.pos < this.input.length
+    ) {
+      ch = this.input.charCodeAt(this.state.pos);
+    }
```

Comment:
Since this code is equal to the one used to parse single-line comments, maybe there should be a `skipLine` method?

Reply:
We could, but I also don't feel that strongly about factoring out every piece of sharable logic.